pages/login.jsx

```diff
@@ -0,0 +1,89 @@
+                        fill="#1E2F97"
+                    />
+                    <path
+                        fillRule="evenodd"
+                        clipRule="evenodd"
+                        d="M3.85756 3.93933C5.82326 3.93933 7.41678 5.48142 7.41678 7.38367V26.4949C7.41678 28.3971 5.82326 29.9392 3.85756 29.9392C1.89186 29.9392 0.29834 28.3971 0.29834 26.4949V7.38367C0.29834 5.48142 1.89186 3.93933 3.85756 3.93933Z"
+                        fill="#1E2F97"
+                    />
+                    <path
+                        fillRule="evenodd"
+                        clipRule="evenodd"
+                        d="M31.1211 7.96186C28.2302 6.21375 24.0029 6.8696 21.4708 10.0697C20.2766 11.5788 18.1034 11.8195 16.6168 10.6073C15.1301 9.39514 14.893 7.18905 16.0871 5.67992C20.6426 -0.0772336 28.6274 -1.70468 34.6542 1.93907C39.7131 4.99673 42.3844 11.2097 41.1125 17.1332C40.2906 20.9595 38.0991 23.5319 36.1317 25.1417C36.1708 25.1987 36.21 25.2558 36.2491 25.3128C37.4892 27.1201 38.7283 28.926 39.9692 30.732C41.06 32.3197 40.6765 34.5044 39.1125 35.6118C37.5484 36.7191 35.3963 36.3298 34.3054 34.7421C33.0633 32.9344 31.8215 31.1245 30.5801 29.3153C29.8493 28.2502 29.1186 27.1854 28.3882 26.1214C27.7897 25.2495 27.6131 24.149 27.9083 23.1294C28.2034 22.1098 28.9383 21.2812 29.9063 20.8766C30.2479 20.7338 33.6052 19.18 34.3656 15.6402M31.1211 7.96186C33.6179 9.47072 35.0072 12.6516 34.3656 15.6402L31.1211 7.96186Z"
+                        fill="#85BDE5"
+                    />
+                    <path
+                        fillRule="evenodd"
+                        clipRule="evenodd"
+                        d="M26.2891 13.7072C25.9228 13.9729 25.7115 14.2525 25.7049 14.2613C25.7047 14.2615 25.7047 14.2616 25.7047 14.2615C25.1862 14.9688 24.0967 15.1819 23.2712 14.7376C22.4457 14.2933 22.197 13.3598 22.7155 12.6525C22.8147 12.5172 23.2513 11.9476 24.0081 11.3987C24.7661 10.8487 25.9675 10.2345 27.5446 10.2424L27.5467 10.2424C28.8312 10.2501 29.801 10.6699 30.3049 10.9355C31.1351 11.3733 31.394 12.3048 30.8831 13.0162C30.3722 13.7276 29.285 13.9493 28.4548 13.5116C28.2449 13.401 27.9168 13.2697 27.523 13.2671C27.0746 13.2651 26.6623 13.4365 26.2891 13.7072Z"
+                        fill="#85BDE5"
+                    />
+                </svg>
+            </h1>
+            <center>
+                <h1 className="text-2xl font-bold mt-0 rounded-sm py-1 px-1 text-blue-800">
+                    <div key="google">
+                        <button onClick={ () => signIn('google', { callbackUrl: "/" }) } className='border-blue-800 border-solid border-2 p-5 rounded-lg'>
```

Comment:
This forces users to go to main page. Try to read the query parameter from http://localhost:3000/login?callbackUrl=%2F and pass that into here. Or there might be a different better way